lang/en/tool_userupsert.php

```diff
@@ -29,18 +29,19 @@
 $string['datamapping'] = 'Use the following options to map data from Web service fields to user profile fields. Any data not mapped to a profile field will be ignored.';
 $string['defaultauth'] = 'Default authentication method';
 $string['defaultauth_desc'] = 'If an authentication method is not provided in Web service data, then this auth method will be set by default.';
-$string['error:customfield'] = 'Error setting custom field data';
-$string['error:deleting'] = 'Error deleting a user';
-$string['error:emailtaken'] = 'Email is already taken: {$a}';
-$string['error:invalidauth'] = 'Invalid auth method: {$a}';
-$string['error:invalidemail'] = 'Invalid email: {$a}';
-$string['error:invalidstatus'] = 'Invalid status: {$a}';
-$string['error:missingfield'] = 'Missing mandatory field {$a}';
+$string['error:customfield'] = 'Error setting custom field data. Error: {$a->error}';
+$string['error:deleting'] = 'Error deleting a user. Error: {$a->error}';
+$string['error:emailtaken'] = 'Email is already taken: {$a->email}';
```

Comment:
Why this needs to be passed as an object? Why not just `{$a}`. It's just one-property object

Reply:
because in some cases there are two properties. E.g. $string['error:notallowedemail'] = 'Email is not allowed: {$a->email}. Error: {$a->error}';

Reply:
Ah, I see. This is because you need two properties in one case, so it has to be an object and therefore the exception cunstructor needs to be tweaked

Reply:
That's correct.